packages/repository/src/query.ts

```diff
@@ -698,6 +698,70 @@ export class FilterBuilder<MT extends object = AnyObject> {
   }
 }
 
+/**
+ * Checks whether fields array passed as an argument is a
+ * subset of fields picked by a target filter.
+ *
+ * @param fields - array of fields to search in a filter
+ * @param filter - target filter
+ *
+ */
+export function matchesFields<T extends object = AnyObject>(
+  fields: (keyof T)[],
+  filter?: Filter<T>,
+) {
+  const normalized = new FilterBuilder(filter).build();
+  const targetFields = normalized.fields;
+  if (!targetFields) {
+    return true;
+  }
+  const isDisablingOnly =
+    _.size(targetFields) > 0 && !_.some(targetFields, Boolean);
+  for (const f of fields) {
+    if (!targetFields[f] && (f in targetFields || !isDisablingOnly)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Ensures that a filter always picks target fields.
+ *
+ * @param targetFields - fields to include
+ * @param filter - a target filter
+ */
```

Comment:
IIUC, `Ensures that a filter always picks target fields.` means 
`queries that apply the returned filter would always include the target fields. If the field in the targetFields is supposed to be returned as undefined, it will be added to the pruneFields for latter processing.` Could you make API docs more descriptive and add the explanation for what is being returned ( `@return ......`)?

Reply:
Fixed.